### DISCUSSION TITLE
adaptive does not set color temperature shelly rgbw cct  Color Temperature Light Mixer

### DISCUSSION THREAD

**Discussion by Rehbinski** [basnijholt/adaptive-lighting#1023] [Q&A]
Please confirm that you have completed the following steps:

 I have updated to the [latest Adaptive Lighting version]
Zigbee2MQTT is disabled

Adaptive-lightning config:

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/e6f625db-41e8-4c74-b05e-656fe4adf04d)

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/4a31941a-1317-4dae-b906-278e4ac9575f)

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/a656deeb-b1c3-44cf-831a-22fe706ecb82)

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/4bdf881f-58cd-408c-93a6-eb3d968b56bc)

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/913c943e-015a-4b27-9409-c904049a9b0d)

Color Temperture Light Mixer

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/82fde6ce-e5de-4a61-bdb4-8704dcf072b5)

automation

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/12264ef9-14a0-486e-a561-14029dcee80f)

testlicht

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/09018711-d81a-49ad-a5a6-22fd0b872943)

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/a162d631-6fc3-41e2-a67d-253a188b78f3)

![image](https://github.com/basnijholt/adaptive-lighting/assets/28094739/694c51ca-83b1-4987-a7eb-bdbfbeea819f)

The problem ist that only the brightnees is changing. The Color temperture is always 2666...

Can anyone help me please?

**Comment by dn-gt** [comment]
Did you found any solution in the meantime? I'm facing the same issue in combination with two Shelly RGBW2 channels configured as virtual light with [Color Temperature Light Mixer](https://github.com/mion00/color-temperature-light-mixer).

**Comment by dn-gt** [comment]
I have tried almost all different kind of combinations in the configuration of Adaptive Lighting. Either the max brightness condition is not considered (one channel at 100% even wit 30% Max Brightness) or color temperature is not correct (typically both channels with same brightness) or both issues together. I even have the feeling it makes a difference if I switch on the virtual device vs. the entity. And it seems if I reload the Integration (Adaptive Lighting , not Color temperature mixer) the brightness adapts to expected value. Any idea would be appreciated.

**Comment by peter1rhodes** [comment]
I have exactly the same issue and it's frustrating!

**Comment by nonamejuju** [comment]
Whit the help of claude I managed to fix the issue with a workaround:

## Root cause found + working workaround

Setup: Shelly RGBW2 with warm + cold LED strips on separate channels,
combined via `color_temperature_light_mixer` into a single
`light.kitchen_light_top` entity, controlled by Adaptive Lighting.

### Root cause

AL's group expansion logic sees the mixer entity as a group and
automatically expands it to its underlying member entities
(the individual Shelly channels). This can be confirmed in debug logs:

    DEBUG [custom_components.adaptive_lighting.switch]
    Expanded light.kitchen_light_top to
    ['light.kitchen_top_led_channel_3', 'light.kitchen_top_led_channel_4']

AL then sends color temperature commands directly to the raw brightness
channels, which have no concept of color temperature and silently ignore
them. The mixer never receives the command, so color temperature never
changes. There is no AL option to disable this expansion behavior.

### Workaround

Remove the mixer entity from the AL `lights:` list entirely.
Drive it instead via an automation that reads AL's computed
`color_temp_kelvin` attribute and applies it directly to the mixer:

    alias: "Kitchen — apply AL color temp via mixer"
    mode: restart
    trigger:
      - platform: time_pattern
        minutes: "/1"
      - platform: state
        entity_id: switch.adaptive_lighting_YOUR_INSTANCE
      - platform: state
        entity_id: light.YOUR_MIXER_ENTITY
        to: "on"
    condition:
      - condition: state
        entity_id: light.YOUR_MIXER_ENTITY
        state: "on"
      - condition: template
        value_template: >
          {{ state_attr('switch.adaptive_lighting_YOUR_INSTANCE', 'color_temp_kelvin') | int
             != state_attr('light.YOUR_MIXER_ENTITY', 'color_temp_kelvin') | int }}
    action:
      - service: light.turn_on
        target:
          entity_id: light.YOUR_MIXER_ENTITY
        data:
          color_temp_kelvin: >
            {{ state_attr('switch.adaptive_lighting_YOUR_INSTANCE', 'color_temp_kelvin') | int }}

The condition on the second `condition` block avoids redundant calls when
the mixer is already at the correct temperature.

A proper fix would require an option in AL to treat an entity as opaque
and skip group expansion, sending commands to the entity itself rather
than resolving its members.

**Reply by peter1rhodes** [reply]
That's great work, thank you for following this up! I had to change my devices because of this issue, but it means I should be able to use my Shelly RGBW2 for something else in the future. A shame it's not fixable within this integration though — is there surely not a way to mimic the inputs AL expects to use?

**Reply by nonamejuju** [reply]
Haven't find any other way unfortunately. This thing was bugging me for a couple of year so finally decided to do something about it. 
May I ask what devices did you end up using ?
I went Shelly + Auxmer light strips because it was the brightest available strip possible with 3600 lumens per meters.

**Comment by peter1rhodes** [comment]
Generic AliExpress 12V strips (either CCT ones with two LED colours, or
individual single-colour strips). The controller I changed to is the
QuinLED An-Penta-Mini. I highly recommend it; fast, flexible ESPHome
control in a small package. You can run WLED as well, if you want, but
personally I've never got on with WLED so I just use ESPHome
